### PR TITLE
ci: Separate pipeline for tensorflow-enabled linux builds

### LINKED ISSUE #1919: Fix run-time `libtensorflow.so` dependency for non-scene-classification node env

### PR DESCRIPTION
**Pull request by jailuthra**
**What does this pull request do? Explain your changes. (required)**

Separate out the build process when `libtensorflow` is enabled.

**Specific updates (required)**

*see commit history*

**How did you test each of these updates (required)**

The CI builds + uploads work as expected.
https://build.livepeer.live/0.5.18-3aa0d8ca/livepeer-linux-tensorflow-amd64.tar.gz <-- Depends on libtensorflow.so - `ldd livepeer | grep tensorflow` 
https://build.livepeer.live/0.5.18-3aa0d8ca/livepeer-linux-amd64.tar.gz <-- Standard build with no runtime dependency
**Does this pull request close any open issues?**

Fixes #1919 

**Checklist:**

- [ ] Read the [contribution guide](./doc/contributing.md)
- [x] `make` runs successfully
- [x] All tests in `./test.sh` pass
- [ ] README and other documentation updated
- [ ] [Pending changelog](./CHANGELOG_PENDING.md) updated

### CODE REVIEW

File: install_ffmpeg.sh
Comment (yondonfu) [inline install_ffmpeg.sh]: Think we want to add `--enable-tensorflow` to `EXTRA_FFMPEG_FLAGS` and not set the var which would overwrite its existing value yeah?

File: install_ffmpeg.sh
Comment (yondonfu) [inline install_ffmpeg.sh:116]: Oh cool does the `*...*` syntax result in a substring check for "experimental" in `BUILD_TAGS`?

File: install_ffmpeg.sh
Comment (yondonfu) [inline install_ffmpeg.sh:138]: Should we conditionally enable `lvpdnn` based on the `experimental` tag as well?

Comment (jailuthra) [reply install_ffmpeg.sh:138]: The `vf_lvpdnn.c` file currently has the definition of the model loading functions, whose declaration is in `avfilter.h` that we always include.
Not enabling this filter lead to compiler errors for the undefined methods.

File: install_ffmpeg.sh
Comment (jailuthra) [reply install_ffmpeg.sh:116]: Yep!

File: install_ffmpeg.sh
Comment (jailuthra) [reply install_ffmpeg.sh]: Ah good catch, my bad.

File: install_ffmpeg.sh
Comment (jailuthra) [reply install_ffmpeg.sh:138]: But agreed, once we land up on a better solution for model loading on the ffmpeg side - we should conditionally include that filter.

Comment (yondonfu) [reply install_ffmpeg.sh:138]: Gotcha thanks for the clarification.

File: install_ffmpeg.sh
Comment (jailuthra) [reply install_ffmpeg.sh]: Fixed in https://github.com/livepeer/go-livepeer/pull/1926/commits/19d2c35db875faab4f0e058afeb86bd94685dca6

File: .github/workflows/linux-tensorflow.yml
Comment (jailuthra) [inline .github/workflows/linux-tensorflow.yml:63]: Decided to explicitly enable the experimental tag rather than using the `ci_env.sh` script as we'd *always* want a tensorflow build here, even for a tagged release.

Comment (yondonfu) [reply .github/workflows/linux-tensorflow.yml:63]: So, this GH action will always build Tensorflow enabled binaries. But, the scene classification capability will only be enabled in the binary for non-semver tagged releases (i.e. an orchestrator would not advertise the capability to a broadcaster even though the binary itself has Tensorflow support). Seems reasonable.

**Review by yondonfu — COMMENTED** [review]
Summary of my current understanding of the CI process:

- Linux
    - Tensorflow Build
        - Always enables Tensorflow. User will need to download libtensorflow themselves to use with the binary
        - Only enables scene classification capability for non-semver tagged releases
    - Regular Build
        - Never enables Tensorflow
        - Only enables scene classification capability for non-semver tagged releases
        - Since Tensorflow is never enabled, scene classification can be enabled for a non-semver tagged release, but will not actually work
 - Docker
     - Always enables Tensorflow and libtensorflow is included in the image
     - Only enables scene classification capability for non-semver tagged releases
 - Windows
     - Never enables Tensorflow
     - Only enables scene classification capability for non-semver tagged releases
     - Since Tensorflow is never enabled, scene classification can be enabled for a non-semver tagged release, but will not actually work
 - Darwin
     - Never enables Tensorflow
     - Only enables scene classification capability for non-semver tagged releases
     - Since Tensorflow is never enabled, scene classification can be enabled for a non-semver tagged release, but will not actually work

**Review by yondonfu — APPROVED** [review]
Just want to make sure we're on the same page about [this comment](https://github.com/livepeer/go-livepeer/pull/1926#pullrequestreview-689817272).

Other than that LGTM after squashing.

**Comment by jailuthra** [comment]
> Summary of my current understanding of the CI process:
> ..

Thanks for the summary, and yeah that sounds about right!
Keeping it functional only for linux + non-semver because the usecase (for now) is limited to our deployments and not to the wider community.

**Pull request merged**
